pystac/extensions/eo.py

```diff
@@ -65,6 +66,13 @@ def get_bands(self, asset=None):
         else:
             bands = self.item.properties.get('eo:bands')
 
+        # get assets with eo:bands even if not in item
+        if asset is None and bands is None:
+            bands = []
+            for (key, value) in self.item.get_assets().items():
+                if 'eo:bands' in value.properties:
+                    bands.extend(value.properties.get('eo:bands'))
```

Comment:
What happens if multiple assets have the same bands? This could be the case if you have for example RGB and RGBIR imagery in the same Item.

Reply:
I see your point. IMO, the `eo:bands` property should be limited to assets because there are strongly linked. Furthermore, the results of get_bands should be a dictionary key/value with key as the asset key and value the list of `eo:bands`. My proposal here is to resolve a case that is allowed by the specification.
I would keep it that way even if leading to 'duplicates' in the function returned list. It simply returns what is specified in the item. If I extend your example, the red band of the RGB could be different of the RGBIR because their composition could be made differently (e.g. different center_wavelength based on optical calibration) and thus that would be 2 different bands.

Reply:
It's a good point about the bands being potentially different but with the same name - band.name isn't unique according to the spec, so really you'd have to check all properties in order to de-duplicate. I do think this is probably a less common case though.

@matthewhanson you've worked with the eo extension a bunch, do you have an opinion here? This is for a pystac.Item returning bands from the extension call `item.ext.eo.get_bands()`, returning either the bands specified in the Item or a aggregation of bands specified in the assets - the potential for duplication happening in the latter case.

Reply:
In lieu of further input and to move this along, I'm in favor of your suggestion to keep it as is, and we can modify if this seems like the incorrect choice down the road.